### PR TITLE
Dynamic CI validation

### PR DESCRIPTION
**Pull request by FynnBe**
todo:
 - [x] update bioimageio.spec (https://github.com/bioimage-io/spec-bioimage-io/pull/262)
 - [x] update bioimageio.core (https://github.com/bioimage-io/core-bioimage-io-python/pull/130)
 - [x] use conda packages in dev/environment-ilastik.yaml
 - [x] add more weights to `validate-ilastik` job
 - [ ] ~~add more validate jobs~~ let's do that later

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    - run: npm ci
+    - run: npm run build
+    - name: "Upload npm output"
+      uses: actions/upload-artifact@v2
+      with:
+        name: npm-dist
+        path: dist
+        retention-days: 1
+
+#  validate-ilastik:
```
Comment (constantinpape) [inline .github/workflows/validate.yml:37]: Does this do anything ilastik specific or is it just testing to run the model with `bioimageio.core`? If so, we should give it a different name.

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      with:
+        path: dist
+    - run: mv dist/npm-dist/* dist/
+    - run: rm -r dist/npm-dist
+    - run: cp ./README.md ./dist/README.md
+    - name: Get branch name to deploy to
+      id: get_branch
+      shell: bash
+      run: |
+        if github.event_name == 'pull_request'; then branch=gh-pages-${{ github.head_ref }}; else branch=gh-pages; fi
```
Comment (FynnBe) [inline .github/workflows/validate.yml:86]: This returns "gh-pages" at the moment, which--here in the PR-- it should not! @constantinpape @oeway Any suggestions how to fix it?

File: .github/workflows/validate.yml (hunk 1 above)
Comment (FynnBe) [reply .github/workflows/validate.yml]: @constantinpape how about dubbing it `bio-core-mix`?

**Comment by FynnBe** [comment]
currently there is no mechanism for checking if a given environment even supports the weight format to be tested, meaning it will go through all models and fail with the same ImportError. I suppose this is fine (at least for the first iteration of this CI).

File: .github/workflows/validate.yml (hunk 1 above)
Comment (constantinpape) [reply .github/workflows/validate.yml]: Sounds good.

**Comment by constantinpape** [comment]
> we can merge this before though, right? Once the next bioimageio.core version is published it will just change the output here; without additional changes here.

Sure, feel free to merge this when you think it's ready/

**Pull request merged**